ledger/src/blockstore.rs

```diff
@@ -1500,19 +1503,43 @@ impl Blockstore {
 
     pub fn read_transaction_status(
         &self,
-        index: (Slot, Signature),
+        index: (Signature, Slot),
     ) -> Result<Option<RpcTransactionStatusMeta>> {
         self.transaction_status_cf.get(index)
     }
 
     pub fn write_transaction_status(
         &self,
-        index: (Slot, Signature),
+        index: (Signature, Slot),
         status: &RpcTransactionStatusMeta,
     ) -> Result<()> {
         self.transaction_status_cf.put(index, status)
     }
 
+    fn prune_transaction_status_cf_by_slot(
+        &self,
+        batch: &mut WriteBatch,
+        from_slot: Slot,
+        to_slot: Slot,
+    ) -> Result<bool> {
+        let mut results: Vec<bool> = vec![];
+        for slot in from_slot..to_slot {
+            let res = self
+                .get_slot_entries(slot, 0, None)?
```

Comment:
So this is kind of an edge case, but this function will fail if any entry in the slot fails to deserialize: https://github.com/solana-labs/solana/blob/master/ledger/src/blockstore.rs#L1615, but ReplayStage may have played some earlier entries in the slot, and stored transaction statuses for those transactions.

Reply:
So for instance imagine:

1. Entry 1 + 2 arrive in blockstore
2. ReplayStage plays entry 1 and entry 2, stores tx statuses
3. Entry 3 arrives which doesn't deserialize

Now here we try to read all 3 entries, and this returns an error, and none of the earlier tx statuses are pruned.

Reply:
Yeah, I follow. Is there a better api to use? Or I need to write a new one that doesn't error out?

Reply:
@CriesofCarrots i think you may be able to adjust the current one to have an option of whether to give you everything before the first error.

Reply:
Unfortunately there's not one that exists that already does this :(